ir/instr.h

```diff
@@ -364,11 +364,25 @@ class Alloc final : public Instr {
   std::unique_ptr<Instr> dup(const std::string &suffix) const override;
 };
 
+class Malloc final : public Instr {
+  Value *size;
+public:
+  Malloc(Type &type, std::string &&name, Value &size)
+    : Instr(type, std::move(name)), size(&size) {}
+
+  std::vector<Value*> operands() const override;
+  void rauw(const Value &what, Value &with) override;
+  void print(std::ostream &os) const override;
+  StateValue toSMT(State &s) const override;
+  smt::expr getTypeConstraints(const Function &f) const override;
+  std::unique_ptr<Instr> dup(const std::string &suffix) const override;
+};
 
 class Free final : public Instr {
+  Type *type;
```

Comment:
why does free() need a type?

Reply:
No special reason, just because free in LLVM IR was taking the type of the pointer operand as well (`call void free(i8* ..)`)
I didn't have special preference, and `free %ptr` seems succinct,  so I updated it